pkg/api/admin/openshiftcluster.go

```diff
@@ -24,27 +24,29 @@ type OpenShiftCluster struct {
-	LastProvisioningState   ProvisioningState       `json:"lastProvisioningState,omitempty"`
-	FailedProvisioningState ProvisioningState       `json:"failedProvisioningState,omitempty"`
-	LastAdminUpdateError    string                  `json:"lastAdminUpdateError,omitempty"`
-	MaintenanceTask         MaintenanceTask         `json:"maintenanceTask,omitempty" mutable:"true"`
-	OperatorFlags           OperatorFlags           `json:"operatorFlags,omitempty" mutable:"true"`
-	OperatorVersion         string                  `json:"operatorVersion,omitempty" mutable:"true"`
-	CreatedAt               time.Time               `json:"createdAt,omitempty"`
-	CreatedBy               string                  `json:"createdBy,omitempty"`
-	ProvisionedBy           string                  `json:"provisionedBy,omitempty"`
-	ClusterProfile          ClusterProfile          `json:"clusterProfile,omitempty"`
-	FeatureProfile          FeatureProfile          `json:"featureProfile,omitempty"`
-	ConsoleProfile          ConsoleProfile          `json:"consoleProfile,omitempty"`
-	ServicePrincipalProfile ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
-	NetworkProfile          NetworkProfile          `json:"networkProfile,omitempty"`
-	MasterProfile           MasterProfile           `json:"masterProfile,omitempty"`
+	ArchitectureVersion             ArchitectureVersion              `json:"architectureVersion"` // ArchitectureVersion is int so 0 is valid value to be returned
+	ProvisioningState               ProvisioningState                `json:"provisioningState,omitempty"`
+	LastProvisioningState           ProvisioningState                `json:"lastProvisioningState,omitempty"`
+	FailedProvisioningState         ProvisioningState                `json:"failedProvisioningState,omitempty"`
+	LastAdminUpdateError            string                           `json:"lastAdminUpdateError,omitempty"`
+	MaintenanceTask                 MaintenanceTask                  `json:"maintenanceTask,omitempty" mutable:"true"`
+	OperatorFlags                   OperatorFlags                    `json:"operatorFlags,omitempty" mutable:"true"`
+	OperatorVersion                 string                           `json:"operatorVersion,omitempty" mutable:"true"`
+	CreatedAt                       time.Time                        `json:"createdAt,omitempty"`
+	CreatedBy                       string                           `json:"createdBy,omitempty"`
+	ProvisionedBy                   string                           `json:"provisionedBy,omitempty"`
+	ClusterProfile                  ClusterProfile                   `json:"clusterProfile,omitempty"`
+	FeatureProfile                  FeatureProfile                   `json:"featureProfile,omitempty"`
+	ConsoleProfile                  ConsoleProfile                   `json:"consoleProfile,omitempty"`
+	ServicePrincipalProfile         ServicePrincipalProfile          `json:"servicePrincipalProfile,omitempty"`
```

Comment:
Are we planning on turning this field into a pointer, to allow it to be nil for WIMI clusters (inverse of `PlatformWorkloadIdentityProfile` below)? 

Making this change would require us to ensure nil checks across all of our references to the field throughout the codebase, so I'm in favor of it being left out of this PR and in a dedicated one if we do go down that path.

Reply:
I'm personally in favor of making that change, and I agree that it should go in a dedicated PR.

Reply:
Agreed, we have talked about doing this as well. I created a dedicated card for this: https://issues.redhat.com/browse/ARO-6623